src/config-watchers.ts

```diff
@@ -3,10 +3,51 @@ import * as vscode from 'vscode';
 import {ClangdContext} from './clangd-context';
 import * as config from './config';
 
+async function promtRestart(settingName: string, promptMessage: string) {
+  switch (config.get<string>(settingName)) {
+  case 'restart':
+    vscode.commands.executeCommand('clangd.restart');
+    break;
+  case 'ignore':
+    break;
+  case 'prompt':
+  default:
+    switch (await vscode.window.showInformationMessage(
+        promptMessage, 'Yes', 'Yes, always', 'No, never')) {
+    case 'Yes':
+      vscode.commands.executeCommand('clangd.restart');
+      break;
+    case 'Yes, always':
+      vscode.commands.executeCommand('clangd.restart');
+      config.update<string>(settingName, 'restart',
+                            vscode.ConfigurationTarget.Global);
+      break;
+    case 'No, never':
+      config.update<string>(settingName, 'ignore',
+                            vscode.ConfigurationTarget.Global);
+      break;
+    default:
+      break;
+    }
+    break;
+  }
+}
+
 export function activate(context: ClangdContext) {
   if (config.get<string>('onConfigChanged') != 'ignore') {
     const watcher = new ConfigFileWatcher(context);
   }
+  vscode.workspace.onDidChangeConfiguration(event => {
+    let Settings: string[] = ['clangd.path', 'clangd.arguments'];
+    Settings.forEach(element => {
```

Comment:
I'd convert this to a for in loop and add a break after the first succesful match, otherwise if multiple settings change at the same time the prompt might be shown repeatedly

Reply:
Good catch.